Sources/CardPayments/APIRequests/CheckoutOrdersAPI.swift

```diff
@@ -39,6 +43,9 @@ class CheckoutOrdersAPI {
         )
         
         let httpResponse = try await apiClient.fetch(request: restRequest)
+        let httpResponseBodyJSON = try? JSONSerialization.jsonObject(with: httpResponse.body ?? Data()) as? [String: Any]
+        correlationID = httpResponseBodyJSON?["debug_id"] as? String
+
         return try HTTPResponseParser().parseREST(httpResponse, as: ConfirmPaymentSourceResponse.self)
```

Comment:
> If it's in the body, this should be included in the call on like 49, and added to ConfirmPaymentSourceResponse model.

Yeah - that's why I mentioned the entire layer would likely need to be rewritten. It's _only_ returned when there is an error, so we don't actually return a `ConfirmPaymentSourceResponse` when a `debug_id` is present since the SDK throws an error as part of the try at that point. Adding it to the response would mean it's always `nil` since we only get a `ConfirmPaymentSourceResponse` on success (and there is no `debug_id` on success according to PPaaS)

Reply:
Maybe in another ticket we can dig into when we expect that value to come in the error body, or in the response headers and which to prefer when 🏈

Reply:
Yeah - If we add it to those thrown errors ([from here](https://github.com/paypal/paypal-ios/blob/0fb35166d971f3001eb2aa22f55d4918fef7d8bb/Sources/CorePayments/Networking/HTTPResponseParser.swift#L58-L70)), we'd end up needing to parse the error again on the client to extract out the debug ID from the thrown error. It could make more sense to return a result object or closure as part of `parseREST` as well as update `CheckoutOrdersAPI` to do the same. Or we can take the same approach as web and just return it with all errors.

It's certainly a bit of an odd series of events and would be a good opportunity to align with Android on! Looking between the two codebases for this certainly illustrated how different this layer is in particular. I'll revert this for now in any case!

Reply:
Reverted: https://github.com/paypal/paypal-ios/pull/208/commits/b031319cbdf8d2133e82a5c9652d7ccd152b68f7